src/platform/packages/private/kbn-grid-layout/grid/use_grid_layout_events/panel/state_manager_actions.test.ts

```diff
@@ -135,6 +137,51 @@ describe('panel state manager actions', () => {
         expect(panel.height).toBe(2);
         expect(panel.width).toBe(2);
       });
+
+      it('can extend panel to the full width of the layout when it is offset', () => {
+        gridLayoutStateManager.layoutRef.current = {
+          getBoundingClientRect: jest.fn().mockReturnValue({
+            top: 0,
+            height: 100,
+            bottom: 100,
+            left: 100, // offset the layout by 100 pixels
+            width: 200,
+            right: 100 + 872, // 872 = the width of the layout when column pixel width is 10
+          }),
+        } as any as HTMLDivElement;
+        gridLayoutStateManager.panelRefs.current.panel1 = {
+          getBoundingClientRect: jest.fn().mockReturnValue({
+            top: 0,
+            left: 100, // offset the panel by 100 pixels to the left as well
+            right: 150,
+            bottom: 50,
+            height: 50,
+            width: 50,
+          }),
+          scrollIntoView: jest.fn(),
+        } as any as HTMLDivElement;
+
+        // update the active panel event
+        gridLayoutStateManager.activePanelEvent$.next({
+          ...(gridLayoutStateManager.activePanelEvent$.getValue() as ActivePanelEvent),
+          panelDiv: gridLayoutStateManager.panelRefs.current.panel1!,
+        });
+
+        // bring the mouse **way** to the right to try to make the panel as big as possible
+        moveAction(
+          new MouseEvent('pointerdown', { clientX: 0, clientY: 0 }),
+          gridLayoutStateManager,
+          {
+            clientX: 10000,
+            clientY: 0,
+          },
+          lastRequestedPanelPosition
+        );
+
+        // the panel reaches the end of the layout
+        const panel = gridLayoutStateManager.gridLayout$.getValue()['main-0'].panels.panel1;
+        expect(panel.width).toBe(48);
```

Comment:
Is there any delay in setting panel width? Should this expect clause be wrapped in `waitFor`?

Reply:
This should be immediate - nothing async is happening with `moveAction`